code/modules/admin/verbs/listahelps.dm

```diff
@@ -0,0 +1,307 @@
+			count++
+
+	if(count < 1)
+		usr << "<b>You don't have any ACTIVE ahelps!</b>"
+		return
+
+
+
+	for(var/datum/adminticket/T in admintickets)
+		if(T.resolved == "No" && T.admin == ckey)
+			usr << "<span class='adminnotice'><b><font color=red>#[T.ID] By:</font> <A HREF='?priv_msg=[T.permckey];ahelp_reply=1'>[key_name(T.permuser)]</A></b><b> Ckey:</b> [T.permckey] <b>Name:</b> [T.permuser] <b>Unique ID:</b> [T.uID]</span>"
+			usr << "	<b>Controls:</b> [ADMIN_QUE(mob)] [ADMIN_PP(mob)] [ADMIN_VV(mob)] [ADMIN_SM(mob)] [ADMIN_FLW(mob)] [ADMIN_TP(mob)]"
+			usr << "	<b>Message:</b> [T.msg]"
+			usr << "	<b>Handling Admin:</b> [T.admin]"
+			usr << "	<b>Replied To:</b> [T.active]/<b><a href='?src=\ref[T];view_logs=\ref[T]'>(LOGS)</a></b>"
+			if(T.resolved == "No")
+				usr << "	<b>Resolved:</b> [T.resolved] <a href='?src=\ref[T];resolve=\ref[T]'>(Resolve)</a>"
+			else
+				usr << "	<b>Resolved:</b> [T.resolved] <a href='?src=\ref[T];resolve=\ref[T]'>(Unresolve)</a>"
+
+/client/verb/viewmyahelp()
+	set category = "Admin"
+	set name = "View my Ahelps"
+	set desc = "List your ahelps"
+
+	var/count = 0
+
+	for(var/datum/adminticket/T in admintickets)
+		if(T.permckey == ckey)
+			count++
+
+	if(count < 1)
+		usr << "<b>You don't have any ahelps!</b>"
+		return
+
+	usr << "<b>Resolved Ahelps</b>"
+
+	var/rpass = 0
+
+	for(var/datum/adminticket/T in admintickets)
+		if(T.permckey == ckey && T.resolved == "Yes")
+			rpass = 1
+			usr << "<span class='adminnotice'><b><font color=red>Adminhelp ID: #[T.ID] </font></b></span>"
+			usr << "	<b>Message:</b> [T.msg]"
+			usr << "	<b>Handling Admin:</b> [T.admin]"
+			usr << "	<b>Replied To:</b> [T.active]/<b><a href='?src=\ref[T];view_logs=\ref[T]'>(LOGS)</a></b>"
+			usr << "	<b>Resolved:</b> [T.resolved]"
+
+	if(rpass == 0)
+		usr << "	None"
+
+	usr << "<b>Unresolved Ahelps</b>"
+
+	var/upass = 0
+
+	for(var/datum/adminticket/T in admintickets)
+		if(T.permckey == ckey && T.resolved == "No")
+			upass = 1
+			usr << "<span class='adminnotice'><b><font color=red>Adminhelp ID: #[T.ID] </font></b></span>"
+			usr << "	<b>Message:</b> [T.msg]"
+			usr << "	<b>Handling Admin:</b> [T.admin]"
+			usr << "	<b>Replied To:</b> [T.active]/<b><a href='?src=\ref[T];view_logs=\ref[T]'>(LOGS)</a></b>"
+			usr << "	<b>Resolved:</b> [T.resolved]"
+
+	if(upass == 0)
+		usr << "	None"
+
+/client/proc/createticket(player, message, uckey, mob)
+	var/datum/adminticket/A = new()
+	A.user = player
+	A.msg = message
+	A.uckey = uckey
+	A.permckey = uckey
+	A.permuser = A.user
+	admintickets += A
+	A.logs += "<b>ADMINHELP:</b> [A.permckey]([A.permuser]): [A.msg]"
+	A.mob = mob
+
+	var/index = 0
+	for(var/datum/adminticket/T in admintickets)
+		index++
+		T.ID = index
+		T.uID = "[T.permckey][T.ID]"
+
+/client/verb/resolveticketself()
+	set category = "Admin"
+	set name = "Resolve My Adminhelp"
+	set desc = "Resolve my own adminhelp"
+
+	var/pass = 0
+	var/datum/adminticket/ticket
+
+	for(var/datum/adminticket/T in admintickets)
```

Comment:
Please use _istypeless for loops_ you can find more information about this in the [contributing.md](https://github.com/HippieStation/HippieStation13/blob/master/CONTRIBUTING.md)